api/anchor.capnp

```diff
@@ -24,8 +24,86 @@ interface Anchor {
     # anchors in the path that do not currently exist are created along
     # the way.
 
+    cell @2 () -> (loader :Loader, storer :Storer, swapper :Swapper);
+    # Cell returns a set of capabilities that provide access to a boxed
+    # value, assigned to the Anchor. Zero or more of these capabilities
+    # MAY be witheld, in which case the corresponding return value will
+    # be null. The loader and storer capabilities respectively map onto
+    # read and write permissions. The swapper is an extended read/write
+    # capability, which SHALL be witheld if either loader or storer are
+    # witheld.
+
     struct Child {
         anchor @0 :Anchor;
         name   @1 :Text;
     }
+
+    struct Value {
+        # Value is a union type that can be assigned to an Anchor.
+        
+        union {
+            null @0 :Void;
+            # Null value indicates that the Anchor is empty, i.e. it
+            # contains no value.
+
+            chan    :union {
+            # Chan values contain some sort of channel. The union is
+            # used as a type hint on the receiving side.
+
+                closer     @1 :import "channel.capnp".Closer;
+                sender     @2 :import "channel.capnp".Sender;
+                recver     @3 :import "channel.capnp".Recver;
+                sendCloser @4 :import "channel.capnp".SendCloser;
+                chan       @5 :import "channel.capnp".Chan;
+            }
+
+            # proc    :group {  # TODO
+            # }
+        }
+    }
+
+    interface Loader {
+        # Loader is a read-only interface to a value.   It grants the
+        # bearer the authority to access an Anchor's underlying value
+        # without modifying it.
+
+        load @0 () -> (value :Value);
+        # Load the Anchor's value atomically.  Note that a concurrent
+        # thread may change the Anchor's value at any point before or
+        # after a call to load.   To conditionally modify an Anchor's
+        # value, use Swapper.CompareAndSwap.
+    }
+
+    interface Storer {
+        # Storer is a write-only interface to a value.  It grants the
+        # bearer the authority to modify an Anchor's underlying value
+        # without accessing its currently-stored value.
+
+        store @0 (value :Value, force :Bool) -> (succeeded :Bool);
```

Comment:
>Removal by writing a null value seems like a good idea!

Yeah, this is shamelessly copied from Circuit's design!  "Remove by writing null" is actually the dual to the "store succeeds only if current value is null" policy.  This turns out to be a nice, small constraint that helps with the common pattern of "set the thing if it's unset", without requiring full-on CAS.  The issue with CAS in this setting is that permission to perform `CompareAndSwap` implies read-write permissions.  With "store only if current value is null", we can do "store if null" in a write-only context, which I find pretty neat 😃 

> I tend to move away from the word “force” unless it necessary

Is this because "force" is somehow ambiguous?  At any rate, I have no objection to "overwrite".  I'll make the change.

Reply:
4ca78cc1431003c1042c4043dba981c2687a062f